src/formats/html.ts

```diff
@@ -75,10 +75,85 @@ export class HtmlImporter extends FormatImporter {
 			return;
 		}
 
+		const fileLookup = new Map<string, { file: PickedFile, tFile: TFile }>;
+		ctx.reportProgress(0, files.length);
 		for (let i = 0; i < files.length; i++) {
 			if (ctx.isCancelled()) return;
 			ctx.reportProgress(i, files.length);
-			await this.processFile(ctx, folder, files[i]);
+			const file = files[i];
+			const tFile = await this.processFile(ctx, folder, file);
+			if (tFile) {
+				fileLookup.set(file instanceof NodePickedFile ? nodeUrl.pathToFileURL(file.filepath).href : file.name, { file, tFile });
+			}
+		}
+
+		if (fileLookup.size > 0) {
+			const { metadataCache } = this.app;
+			// @ts-ignore
+			if (!metadataCache.computeMetadataAsync) {
+				const aFile = fileLookup.values().next().value;
+				const resolved = new Promise<void>(resolve => {
+					const ref = this.app.metadataCache.on('resolved', () => {
+						this.app.metadataCache.offref(ref);
+						resolve();
+					});
+				});
+				const appended = '\n';
+				await this.app.vault.append(aFile, appended);
+				await resolved;
+				await this.app.vault.process(aFile, data => data.endsWith(appended) ? data.slice(0, -appended.length) : data);
```

Comment:
This block of code in the `if` branch ensures that metadata cache for the newly createdfiles is up-to-date by waiting for `resolved`. Since `resolved` is triggered only once the metadata cache is up-to-date after a file modification, if `resolved` triggered by creating the new files is fired before `await resolved`, then `await resolved` will block forever. Making a file change here ensures `resolved` is triggered.

Anyway, I have thought of another way of not needing to append an newline and made this change in 280f148eba9c6e9edec453f6a4b67ade02967c8f.